Apps/Src/ReadCarCAN.c

```diff
@@ -265,25 +241,30 @@ void Task_ReadCarCAN(void *p_arg) {
                 // function instead
                 setMotorControllerContactor(MOTOR_SENSE_ACTUAL_VALUE(dataBuf.data), true);
 
-                Status_Leds_Write(CONTROLS_FAULT, true);
                 // Update Array Precharge sense state
                 Contactors_Set(ARRAY_PRECHARGE_BYPASS_CONTACTOR,
                                ARRAY_PRECHARGE_ACTUAL_VALUE(dataBuf.data), true);
                 // Update Motor Precharge sense state
                 Contactors_Set(MOTOR_CONTROLLER_PRECHARGE_BYPASS_CONTACTOR,
                                MOTOR_PRECHARGE_ACTUAL_VALUE(dataBuf.data), true);
+                
+                bool motorContactorsOn = Contactors_Get(MOTOR_CONTROLLER_CONTACTOR, false) && Contactors_Get(MOTOR_CONTROLLER_PRECHARGE_BYPASS_CONTACTOR, false);
+                bool bps_state = (Contactors_Get(HV_MINUS_CONTACTOR, false) && Contactors_Get(HV_PLUS_CONTACTOR, false));
 
-                if (Contactors_Get(MOTOR_CONTROLLER_CONTACTOR, true) &&
-                    Contactors_Get(MOTOR_CONTROLLER_PRECHARGE_BYPASS_CONTACTOR, true)) {
+                if (motorContactorsOn && bps_state) {
+                    // TODO: might need to check if the motor is already safe to run?
                     motorPrechargeOnCount++;
                     if (motorPrechargeOnCount >= MOTOR_PRECHARGE_ON_COUNT_THRESHOLD) {
                         // If the motor precharge contactor has been on for enough iterations,
                         // we can consider it safe to run
                         MotorStatus_ModifyBits(MOTOR_SAFE_TO_RUN, true, !OS_FLAG_SCHED_POINT);
-                        // OS_ERR err;
-                        // OSFlagPost(&BPS_Motor_Status_Flags, MOTOR_SAFE_TO_RUN, OS_OPT_POST_FLAG_SET, &err);
-                        // assertOSError(err);
+                        motorPrechargeOnCount = 0; // Reset the counter so we don't overflow and/or constantly post to the flag
                     }
+                    // Leave the motor safe to run bit as if it was set
+                }
+                else{
+                    MotorStatus_ModifyBits(MOTOR_SAFE_TO_RUN, false, !OS_FLAG_SCHED_POINT);
+                    motorPrechargeOnCount = 0;
```

Comment:
Won't this keep resetting motorPrechargeOnCount? It's in the else condition so whenever motorPrechargeOnCount < threshold, it will be reset to 0, meaning it never makes progress